src/nbl/video/CVulkanCommandBuffer.h

```diff
@@ -1052,20 +1052,36 @@ class CVulkanCommandBuffer : public IGPUCommandBuffer
         if (layout->getAPIType() != EAT_VULKAN)
             return false;
 
-        constexpr uint32_t MAX_DESCRIPTOR_SET_COUNT = 100u;
+        constexpr uint32_t MAX_DESCRIPTOR_SET_COUNT = 4u;
 
         VkPipelineLayout vk_pipelineLayout = static_cast<const CVulkanPipelineLayout*>(layout)->getInternalObject();
 
-        VkDescriptorSet vk_descriptorSets[MAX_DESCRIPTOR_SET_COUNT];
+        uint32_t dynamicOffsetCountPerSet[MAX_DESCRIPTOR_SET_COUNT] = {};
+
+        VkDescriptorSet vk_descriptorSets[MAX_DESCRIPTOR_SET_COUNT] = {};
         for (uint32_t i = 0u; i < descriptorSetCount; ++i)
         {
             if (pDescriptorSets[i] && pDescriptorSets[i]->getAPIType() == EAT_VULKAN)
+            {
                 vk_descriptorSets[i] = static_cast<const CVulkanDescriptorSet*>(pDescriptorSets[i])->getInternalObject();
+
+                if (dynamicOffsets) // count dynamic offsets per set, if there are any
+                {
+                    auto bindings = pDescriptorSets[i]->getLayout()->getBindings();
+                    for (const auto& binding : bindings)
+                    {
+                        if ((binding.type == asset::EDT_STORAGE_BUFFER_DYNAMIC) || (binding.type == asset::EDT_UNIFORM_BUFFER_DYNAMIC))
+                            dynamicOffsetCountPerSet[i] += binding.count;
+                    }
+                }
+            }
         }
 
         const auto* vk = static_cast<const CVulkanLogicalDevice*>(getOriginDevice())->getFunctionTable();
 
         // Will bind [first, last) with one call
+        uint32_t dynamicOffsetsBindOffset = 0u;
+        uint32_t bindCount = 0u;
```

Comment:
@nipunG314 this function got completely rewritten since then, you can discard all changes to this file